### PR TITLE
Update multi_a to not include n wrapper

### PR DESCRIPTION
**Pull request by sanket1729**
While we are at it, do not use the default impl. This makes these types
of errors easy to make. I think this could have easily been prevented if
I was forced to explicitly write AnyNonZero in front of multi_a.

Reported by darosior.

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK pr/525/head

**Comment by apoelstra** [comment]
There is some github weirdness going on. ACK 7118c848

**Comment by apoelstra** [comment]
I can't merge because Github is not publishing the tip of this PR under the right ref name.

File: tests/test_desc.rs [hunk 1]
```diff
@@ -395,6 +395,10 @@ fn test_descs(cl: &Client, testdata: &TestData) {
     let result = test_desc_satisfy(cl, testdata, "tr(X!,{pk(X1!),pk(X2!)})");
     assert_eq!(result, Err(DescError::PsbtFinalizeError));
 
+    // Test 10: Test taproot desc with ZERO known keys
```
Comment (darosior) [inline tests/test_desc.rs:398]: Comment should be updated?

**Review by darosior — APPROVED** [review]
utACK 7118c8488bc7e2c6219163648dd807c815d744f2

**Pull request merged**